webapp-django/crashstats/crashstats/tests/test_utils.py

```diff
@@ -4,13 +4,53 @@
 
 import copy
 import json
+import os
+import os.path
 
+import pytest
+
+from django.conf import settings
 from django.http import HttpResponse
 from django.utils.encoding import smart_text
 
 from crashstats.crashstats import utils
 
 
+def get_product_details_files():
+    product_details_path = os.path.join(settings.SOCORRO_ROOT, 'product_details')
+    return [
+        os.path.join(product_details_path, fn)
+        for fn in os.listdir(product_details_path)
+        if fn.endswith('.json')
+    ]
+
+
+@pytest.mark.parametrize('fn', get_product_details_files())
+def test_product_details_files(fn):
+    """Validate product_details/ JSON files."""
+    fn_basename = os.path.basename(fn)
+
+    try:
+        with open(fn, 'r') as fp:
+            data = json.load(fp)
+    except json.decoder.JSONDecoderError as exc:
+        raise Exception('%s: invalid JSON: %s' % (fn_basename, exc))
+
+    if 'active_versions' not in data:
+        raise Exception('%s: missing "active_versions" key' % fn_basename)
+
+    if not isinstance(data['active_versions'], list):
+        raise Exception(
+            '%s: "active_versions" value is not a list of strings' % fn_basename
+        )
+
+    for item in data['active_versions']:
+        if not isinstance(item, str):
+            raise Exception(
+                '%s: value %r is not a str' % (fn_basename, item)
+            )
```

Comment:
These tests are designed to make it clearer to the contributor and the reviewer exactly what the problem is with minimal time-wastage.

Reply:
If this file grows in responsibility and other things, we should build a schema and validate it against the schema. I figured I wouldn't do that, yet, but it's a good next step if we do more with these files.